### PR TITLE
[Metrics] Kubernetes Support & Maintenance 

### PR DESCRIPTION
**Pull request by simon-mo**
This PR replace #411 with cleaner commit history and different integration test.

The integration test issue come with #411 is hard to debug. It turns out to be:
- Inside each pod, the kubelet needs time to setup the network service within a pod. We just need to wait for the network service to be setup. It takes 30 seconds to 2 minutes. This is fixed by using retry and while loop inside integration test for kubernetes metrics. 

This PR also adds some maintenance:
- Modify the docker integration test as well to test for specific node health. It turns out the original test 1 calls API that returns the same information not matter the node is healthy or not. 
- ~~Let frontend-exporter to be more fault tolerant with some try-catch. Not a big fix so it just with this PR.~~ Independent PR in #429
- Update readme

### CODE REVIEW

**Comment by simon-mo** [comment]
~~Blocked by #429~~

File: clipper_admin/clipper_admin/kubernetes/prom_deployment.yml
Comment (dcrankshaw) [inline clipper_admin/clipper_admin/kubernetes/prom_deployment.yml]: Let's use a versioned image rather than `latest`

File: clipper_admin/clipper_admin/kubernetes/kubernetes_metric_utils.py
Comment (dcrankshaw) [inline clipper_admin/clipper_admin/kubernetes/kubernetes_metric_utils.py]: Change these file suffixes to `.yaml` to keep them consistent with the rest of the K8s yaml files

File: integration-tests/clipper_metric_kube.py
Comment (dcrankshaw) [inline integration-tests/clipper_metric_kube.py:3]: @simon-mo would it make sense to just add this test to the existing Kubernetes integration test, or do you think it's better to keep it separate? I'll defer to your judgment here.

File: monitoring/README.md
Comment (dcrankshaw) [inline monitoring/README.md]: change `RPC.py` to `rpc.py`

File: monitoring/front_end_exporter.py
Comment (dcrankshaw) [inline monitoring/front_end_exporter.py]: Change print to a logging statement

File: monitoring/front_end_exporter.py
Comment (dcrankshaw) [inline monitoring/front_end_exporter.py]: change print to a logging statement

File: monitoring/front_end_exporter.py
Comment (dcrankshaw) [inline monitoring/front_end_exporter.py]: For the sake of being explicit, change this to `if len(metrics) == 0:`

**Review by dcrankshaw — CHANGES_REQUESTED** [review]
Looks great. Just some minor formatting comments.

File: integration-tests/clipper_metric_kube.py
Comment (simon-mo) [reply integration-tests/clipper_metric_kube.py:3]: The metrics take about 3 minutes. I would keep them separate for now. But this can be a potential solution to speedup our integration test even though it makes the test less modular.

**Comment by simon-mo** [comment]
Jenkins test this please

**Comment by simon-mo** [comment]
jenkins test this please

File: integration-tests/clipper_metric_kube.py
Comment (dcrankshaw) [inline integration-tests/clipper_metric_kube.py]: Is this TODO still outstanding?

**Review by dcrankshaw — APPROVED** [review]
lgtm

**Pull request merged**
